packages/core/src/messagingProtocols/proteus/ProteusService/ProteusService.ts

```diff
@@ -266,7 +277,34 @@ export class ProteusService {
     return qualifiedOTRRecipients;
   }
 
-  public async proteusCryptoboxMigrate(storeName: string) {
-    return this.coreCryptoClient.proteusCryptoboxMigrate(storeName);
+  private async migrateToCoreCrypto(storeEngine: CRUDEngine, context: Context) {
+    const dbName = storeEngine.storeName;
+    const migrationFlag = `${dbName}-corecrypto-ready`;
+    const localStorage = getLocalStorage();
+    if (localStorage.getItem(migrationFlag)) {
+      return;
+    }
+    // We want sessions to be fully qualified from now on
+    migrateToQualifiedSessionIds(storeEngine, context.domain ?? '');
+
+    this.logger.log(`Migrating data from cryptobox store (${dbName}) to corecrypto.`);
+    try {
+      await this.coreCryptoClient.proteusCryptoboxMigrate(dbName);
+
+      // We can clear 3 stores (keys - local identity, prekeys and sessions) from wire db.
+      // They will be stored in corecrypto database now.
+      /* TODO uncomment this code when we are sure migration for wire.com has happened successfully for enough users
```

Comment:
are we creating a follow-up ticket or some tracking for this topic?

Reply:
Created 👍